## Lecture 9 Notes

### Packages used in this notebook

In [8]:
import pandas as pd
from scipy import stats

### Import data

In [9]:
# Import the heart failure dataset
df = pd.read_csv('https://raw.githubusercontent.com/juanklopper/TutorialData/main/heart_failure.csv')

### Analyzing the data

In [10]:
df.head()

age anaemia  creatinine_phosphokinase diabetes  ejection_fraction  \
0  75.0      No                       582       No                 20   
1  55.0      No                      7861       No                 38   
2  65.0      No                       146       No                 20   
3  50.0     Yes                       111       No                 20   
4  65.0     Yes                       160      Yes                 20   

  hypertension  platelets  serum_creatinine  serum_sodium     sex smoking  \
0          Yes  265000.00               1.9           130    Male      No   
1           No  263358.03               1.1           136    Male      No   
2           No  162000.00               1.3           129    Male     Yes   
3           No  210000.00               1.9           137    Male      No   
4           No  327000.00               2.7           116  Female      No   

   time death  
0     4   Yes  
1     6   Yes  
2     7   Yes  
3     7   Yes  
4     8   Yes

In [13]:
df.columns


Index(['age', 'anaemia', 'creatinine_phosphokinase', 'diabetes',
       'ejection_fraction', 'hypertension', 'platelets', 'serum_creatinine',
       'serum_sodium', 'sex', 'smoking', 'time', 'death'],
      dtype='object')

In [14]:
df.shape

(299, 13)

In [15]:
df.info

<bound method DataFrame.info of       age anaemia  creatinine_phosphokinase diabetes  ejection_fraction  \
0    75.0      No                       582       No                 20   
1    55.0      No                      7861       No                 38   
2    65.0      No                       146       No                 20   
3    50.0     Yes                       111       No                 20   
4    65.0     Yes                       160      Yes                 20   
..    ...     ...                       ...      ...                ...   
294  62.0      No                        61      Yes                 38   
295  55.0      No                      1820       No                 38   
296  45.0      No                      2060      Yes                 60   
297  45.0      No                      2413       No                 38   
298  50.0      No                       196       No                 45   

    hypertension  platelets  serum_creatinine  serum_sodium     sex

In [16]:
# determine the unique classes in the dataset for the column anemia column
df['anaemia'].unique()

array(['No', 'Yes'], dtype=object)

In [17]:
# Determince the value counts for the anemia column
df['anaemia'].value_counts()

anaemia
No     170
Yes    129
Name: count, dtype: int64

### 170 people said no to having anemia and 129 said yes

In [18]:
# Deternime the relative frequency of the anemia column
df['anaemia'].value_counts(normalize=True)

anaemia
No     0.568562
Yes    0.431438
Name: proportion, dtype: float64